io/www/src/fragment/wcs.js

```diff
@@ -31,7 +31,12 @@ async function computeCache(tokens, wcsContext) {
                 const response = await fetchArtifact(osi, promotionCode, wcsContext);
                 if (response) {
                     const { resolvedOffers } = response;
-                    const cacheKey = [osi, wcsContext.country.toLowerCase(), wcsContext.language?.toLowerCase(), promotionCode]
+                    const cacheKey = [
+                        osi,
+                        wcsContext.country.toLowerCase(),
+                        wcsContext.language?.toLowerCase(),
+                        promotionCode?.toLowerCase(),
```

Comment:
you should not change `promotionCode` case. It can contain characters in both lower and upper case.
wcs.js in the frontend uses them without as is without changing the case.